src/pair.py

```diff
@@ -5,9 +5,127 @@
 """Pair potentials."""
 
 from hoomd.azplugins import _azplugins
-from hoomd.data.parameterdicts import TypeParameterDict
+from hoomd.data.parameterdicts import ParameterDict, TypeParameterDict
 from hoomd.data.typeparam import TypeParameter
 from hoomd.md import pair
+from hoomd.variant import Variant
+
+
+class DPDGeneralWeight(pair.Pair):
+    r"""Dissipative Particle Dynamics with generalized weight function.
+
+    Args:
+        nlist (hoomd.md.nlist.NeighborList): Neighbor list
+        kT (`hoomd.variant` or `float`): Temperature of
+            thermostat :math:`[\mathrm{energy}]`.
+        default_r_cut (float): Default cutoff radius :math:`[\mathrm{length}]`.
+
+    :py:class:`general` specifies that a DPD pair force should be applied between every
+    non-excluded particle pair in the simulation, including an interaction potential,
+    pairwise drag force, and pairwise random force. The form of the forces between
+    pairs of particles is:
+
+    .. math::
+        :nowrap:
+        \begin{eqnarray*}
+        \mathbf{F} = \mathbf{F}_{\rm C} + \mathbf{F}_{\rm D} +  \mathbf{F}_{\rm R} \\
+        \end{eqnarray*}
+
+    The conservative force :math:`\mathbf{F}_{\rm C}` is the standard form:
+
+    .. math::
+        :nowrap:
+        \begin{eqnarray*}
+        \mathbf{F}_{\rm C} =& A (1- r_{ij}/r_{\rm cut}) & r \le r_{\rm cut} \\
+                           =& 0 & r > r_{\rm cut}
+        \end{eqnarray*}
+
+    where *A* is the interaction parameter and :math:`r_{\rm cut}` is the cutoff radius.
+    Here, :math:`r_{ij} = r_i - r_j`. See `Groot and Warren 1997
+      <http://dx.doi.org/10.1063/1.474784>`_
+    for more details.
+
+    The dissipative and random forces, respectively, are:
+
+    .. math::
+        :nowrap:
+        \begin{eqnarray*}
+        \mathbf{F}_{\rm D} =& -\gamma \omega_{\rm D}(r_{ij})
+          (\mathbf{v}_{ij} \cdot \mathbf{\hat r}_{ij}) \mathbf{\hat r}_{ij} \\
+        \mathbf{F}_{\rm R} =& \sigma \omega_{\rm R}(r_{ij}) \xi_{ij}
+            \mathbf{\hat r}_{ij}
+        \end{eqnarray*}
+
+    where :math:`\sigma = 2\gamma k_{\rm B}T` and
+    :math:`\omega_{\rm D} = \left[\omega_{\rm R} \right]^2`
+    to satisfy the fluctuation dissipation relation. The genealized weight
+    function is given by the form proposed by
+    `Fan et al. <https://doi.org/10.1063/1.2206595>`_:
```

Comment:
I think this needs to be expanded based on  `Fan et al. <https://doi.org/10.1063/1.2206595>`. Also, I think the correct relation between sigma and gamma is :math:`\sigma ^2 = 2\gamma k_{\rm B}T` :
```
where :math:`\sigma` and :math:`gamma` are two coefficients characterizing the strengths of the dissipative and random forces. To satisfy the fluctuation dissipation relation, :math:`\omega_{\rm D} = \left[\omega_{\rm R} \right]^2` where  :math:`\sigma ^2 = 2\gamma k_{\rm B}T`. Here, :math:`v_{ij} = v_i - v_j`, and :math:`\xi_{ij}` is a white noise function. The generalized weight function is given by the form proposed by
`Fan et al. <https://doi.org/10.1063/1.2206595>`_: 
```